### PR TITLE
[Tutorials] Add LLaMa 3.1 405B to the SDG tutorial

### PR DESCRIPTION
**Pull request by Maghoumi**
## Description
* Add suppurt for SDG using LLaMa 3.1 405B
* Update instructions
* Improve error handling
* Update some of the README.md files throughout the repo

## Checklist
- [X] I am familiar with the [Contributing Guide](https://github.com/NVIDIA/NeMo-Curator/blob/main/CONTRIBUTING.md).
- [X] New or Existing tests cover these changes.
- [X] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by Maghoumi** [comment]
Hi @ryantwolf, could you prioritize reviewing this PR? There are some blogs/tutorials going out soon that are going to reference this tutorial.

Thanks!

File: tutorials/peft-curation-with-sdg/README.md [hunk 1]
```diff
@@ -44,9 +44,18 @@ python tutorials/peft-curation-with-sdg/main.py \
...
     --api-key YOUR_BUILD.NVIDIA.COM_API_KEY \
-    --device gpu \
-    --synth-gen-rounds 2 \ # Do 2 rounds of synthetic data generation
-    --synth-gen-ratio 0.5  # Generate synthetic data using 50% of the real data
+    --device gpu \  # Use the GPU and enable semantic deduplication
+    --synth-gen-rounds 1 \ # Do 1 round of synthetic data generation
+    --synth-gen-ratio 0.001 \  # Generate synthetic data using 0.1% of the real data
+    --synth-gen-model "meta/llama-3.1-405b-instruct" # Use LLaMa 3.1 405B
+
+# To control the amount of synthetic data to generate
```
Comment (ryantwolf) [inline tutorials/peft-curation-with-sdg/README.md:52]: Nit: Could you modify this comment and the comment above to reference the model that the command is using? Something like
```python
# To control the amount of synthetic data to generate using Llama 3.1 405B
```
and
```python
# To control the amount of synthetic data to generate using Nemotron-4 340B
```

**Review by ryantwolf — CHANGES_REQUESTED** [review]
Just one nit. Looks good otherwise.

File: tutorials/peft-curation-with-sdg/README.md (hunk 1 above)
Comment (Maghoumi) [reply tutorials/peft-curation-with-sdg/README.md]: Done.

**Review by ryantwolf — APPROVED** [review]

**Pull request merged**